Abseil-cpp wrap doesn't compile on ARM with gcc only

I'm facing some issue with the `abseil-cpp` wrap during compiling in `aarch64` architecture. The issue is is only happening when building with `gcc`. `clang` seems to be OK with it.

I'm also tested compiling `abseil-cpp` without meson wrap (just using `cmake`) and works fine in both compilers.

Check https://github.com/inglor/abseil-wrap for a quick project to reproduce.

Issue:
```
$ meson build
The Meson build system
Version: 0.57.1
Source dir: /home/inglor/cowerPkg/abseil-wrap
Build dir: /home/inglor/cowerPkg/abseil-wrap/build
Build type: native build
Project name: abseil-wrap
Project version: 0
C++ compiler for the host machine: c++ (gcc 10.2.0 "c++ (GCC) 10.2.0")
C++ linker for the host machine: c++ ld.bfd 2.35
Host machine cpu family: aarch64
Host machine cpu: aarch64
Compiler for C++ supports arguments -ffunction-sections: YES
Compiler for C++ supports arguments -fdata-sections: YES
Compiler for C++ supports arguments -Wnon-virtual-dtor: YES
Compiler for C++ supports link arguments -Wl,--gc-sections: YES
Downloading abseil-cpp source from https://github.com/abseil/abseil-cpp/archive/20200923.2.tar.gz
Downloading file of unknown size.
Downloading abseil-cpp patch from https://wrapdb.mesonbuild.com/v1/projects/abseil-cpp/20200923.2/1/get_zip
Download size: 2954
Downloading: ..........

|Executing subproject abseil-cpp method meson
|
|Project name: abseil-cpp
|Project version: 20200923.2
|C++ compiler for the host machine: c++ (gcc 10.2.0 "c++ (GCC) 10.2.0")
|C++ linker for the host machine: c++ ld.bfd 2.35
|Compiler for C++ supports arguments -Wno-sign-compare: YES
|Checking if "GCC atomic builtins" links: YES
|Run-time dependency threads found: YES
|Found pkg-config: /usr/bin/pkg-config (1.7.3)
|Found CMake: /usr/bin/cmake (3.19.6)
|Run-time dependency corefoundation found: NO (tried pkgconfig and cmake)
|Build targets in project: 12
|Subproject abseil-cpp finished.

Build targets in project: 13

abseil-wrap 0

  Subprojects
    abseil-cpp: YES

Found ninja-1.10.2 at /usr/bin/ninja
$ ninja -C build
ninja: Entering directory `build'
[37/123] Compiling C++ object subprojects/abseil-cpp-20200923.2/libabsl_strings_lib.a.p/absl_strings_internal_str_format_bind.cc.o
In file included from ../subprojects/abseil-cpp-20200923.2/absl/strings/internal/str_format/arg.h:18,
                 from ../subprojects/abseil-cpp-20200923.2/absl/strings/internal/str_format/bind.h:10,
                 from ../subprojects/abseil-cpp-20200923.2/absl/strings/internal/str_format/bind.cc:1:
../subprojects/abseil-cpp-20200923.2/absl/strings/internal/str_format/extension.h: In static member function ‘static void absl::lts_2020_09_23::str_format_internal::FormatConversionSpecImplFriend::SetFlags(absl::lts_2020_09_23::str_format_internal::Flags, absl::lts_2020_09_23::str_format_internal::FormatConversionSpecImpl*)’:
../subprojects/abseil-cpp-20200923.2/absl/strings/internal/str_format/extension.h:295:15: note: parameter passing for argument of type ‘absl::lts_2020_09_23::str_format_internal::Flags’ changed in GCC 9.1
  295 |   static void SetFlags(Flags f, FormatConversionSpecImpl* conv) {
      |               ^~~~~~~~
[68/123] Compiling C++ object subprojects/abseil-cpp-20200923.2/libabsl_random_lib.a.p/absl_random_internal_randen_hwaes.cc.o
FAILED: subprojects/abseil-cpp-20200923.2/libabsl_random_lib.a.p/absl_random_internal_randen_hwaes.cc.o
c++ -Isubprojects/abseil-cpp-20200923.2/libabsl_random_lib.a.p -Isubprojects/abseil-cpp-20200923.2 -I../subprojects/abseil-cpp-20200923.2 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -g -Wno-sign-compare -fPIC -MD -MQ subprojects/abseil-cpp-20200923.2/libabsl_random_lib.a.p/absl_random_internal_randen_hwaes.cc.o -MF subprojects/abseil-cpp-20200923.2/libabsl_random_lib.a.p/absl_random_internal_randen_hwaes.cc.o.d
-o subprojects/abseil-cpp-20200923.2/libabsl_random_lib.a.p/absl_random_internal_randen_hwaes.cc.o -c ../subprojects/abseil-cpp-20200923.2/absl/random/internal/randen_hwaes.cc
In file included from ../subprojects/abseil-cpp-20200923.2/absl/random/internal/randen_hwaes.cc:229:
/usr/lib/gcc/aarch64-unknown-linux-gnu/10.2.0/include/arm_neon.h: In function ‘Vector128 {anonymous}::AesRound(const Vector128&, const Vector128&)’:
/usr/lib/gcc/aarch64-unknown-linux-gnu/10.2.0/include/arm_neon.h:12321:1: error: inlining failed in call to ‘always_inline’ ‘uint8x16_t vaesmcq_u8(uint8x16_t)’: target specific option mismatch
12321 | vaesmcq_u8 (uint8x16_t data)
      | ^~~~~~~~~~
../subprojects/abseil-cpp-20200923.2/absl/random/internal/randen_hwaes.cc:255:20: note: called from here
  255 |   return vaesmcq_u8(vaeseq_u8(state, uint8x16_t{})) ^ round_key;
      |          ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../subprojects/abseil-cpp-20200923.2/absl/random/internal/randen_hwaes.cc:229:
/usr/lib/gcc/aarch64-unknown-linux-gnu/10.2.0/include/arm_neon.h:12307:1: error: inlining failed in call to ‘always_inline’ ‘uint8x16_t vaeseq_u8(uint8x16_t, uint8x16_t)’: target specific option mismatch
12307 | vaeseq_u8 (uint8x16_t data, uint8x16_t key)
      | ^~~~~~~~~
../subprojects/abseil-cpp-20200923.2/absl/random/internal/randen_hwaes.cc:255:20: note: called from here
  255 |   return vaesmcq_u8(vaeseq_u8(state, uint8x16_t{})) ^ round_key;
      |          ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[73/123] Compiling C++ object subprojects/abseil-cpp-20200923.2/libabsl_time_lib.a.p/absl_time_duration.cc.o
ninja: build stopped: subcommand failed.
```

--- a/meson.build
+++ b/meson.build
@@ -10,6 +10,15 @@ cpp = meson.get_compiler('cpp')
 cpp_flags = [
   '-Wno-sign-compare'
 ]
+
+if host_machine.cpu_family() == 'x86_64'
+  cpp_flags += [ '-maes', '-msse4.1' ]
+elif host_machine.cpu_family() == 'aarch64' and cpp.sizeof('void*') == 8
+  cpp_flags += [ '-march=armv8-a+crypto' ]
+elif host_machine.cpu_family() == 'arm' and cpp.sizeof('void*') == 4
+  cpp_flags += [ '-mfpu=neon' ]
+endif
+
 flags = cpp.get_supported_arguments(cpp_flags)
 
 add_project_arguments(flags, language: 'cpp')

